src/daft-local-execution/src/sinks/commit_write.rs

```diff
@@ -67,18 +74,47 @@ impl BlockingSink for CommitWriteSink {
         states: Vec<Self::State>,
         spawner: &ExecutionTaskSpawner,
     ) -> BlockingSinkFinalizeResult<Self> {
+        let data_schema = self.data_schema.clone();
         let file_schema = self.file_schema.clone();
         let file_info = self.file_info.clone();
         spawner
             .spawn(
                 async move {
-                    let written_file_path_record_batches = states
+                    let mut written_file_path_record_batches = states
                         .into_iter()
                         .flat_map(|mut state| {
                             std::mem::take(&mut state.written_file_path_record_batches)
                         })
                         .collect::<Vec<_>>();
 
+                    // If nothing was written, create an empty file for Parquet/CSV when not partitioned
+                    if written_file_path_record_batches.is_empty()
+                        && file_info.partition_cols.is_none()
```

Comment:
As mentioned in https://github.com/Eventual-Inc/Daft/issues/5700, for write empty data with partition column, we don't create the empty file any more. But be carefully, will meet FileNotFoundError when during the data after write